From 453c12d732b19c8a23a40b6cc234858c40d58fcd Mon Sep 17 00:00:00 2001
From: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
Date: Thu, 4 Jan 2018 16:35:59 -0600
Subject: [PATCH 0803/1240] ASoC: Intel: boards: align/fix SKL/BXT/KBL Kconfigs

No reason why SND_SOC_INTEL_SST should be set here.
Also make sure same dependencies are used everywhere (only last one has SPI
in addition). Replace X86_INTEL_LPSS by MFD_INTEL_LPSS since the former
makes no sense for Skylake+ devices

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
Reviewed-by: Andy Shevchenko <andriy.shevchenko@linux.intel.com>
Acked-By: Vinod Koul <vinod.koul@intel.com>
Signed-off-by: Mark Brown <broonie@kernel.org>
(cherry picked from commit 969eaef7710400e39be13190bace40910555d426)
---
 sound/soc/intel/boards/Kconfig | 22 ++++++++--------------
 1 file changed, 8 insertions(+), 14 deletions(-)

diff --git a/sound/soc/intel/boards/Kconfig b/sound/soc/intel/boards/Kconfig
index c8b2667..e51585b 100644
--- a/sound/soc/intel/boards/Kconfig
+++ b/sound/soc/intel/boards/Kconfig
@@ -178,7 +178,7 @@ if SND_SOC_INTEL_SKYLAKE
 
 config SND_SOC_INTEL_SKL_RT286_MACH
 	tristate "ASoC Audio driver for SKL with RT286 I2S mode"
-	depends on X86 && ACPI && I2C
+	depends on MFD_INTEL_LPSS && I2C && ACPI
 	select SND_SOC_RT286
 	select SND_SOC_DMIC
 	select SND_SOC_HDAC_HDMI
@@ -190,7 +190,7 @@ config SND_SOC_INTEL_SKL_RT286_MACH
 
 config SND_SOC_INTEL_SKL_NAU88L25_SSM4567_MACH
 	tristate "ASoC Audio driver for SKL with NAU88L25 and SSM4567 in I2S Mode"
-	depends on X86_INTEL_LPSS && I2C
+	depends on MFD_INTEL_LPSS && I2C && ACPI
 	select SND_SOC_NAU8825
 	select SND_SOC_SSM4567
 	select SND_SOC_DMIC
@@ -203,7 +203,7 @@ config SND_SOC_INTEL_SKL_NAU88L25_SSM4567_MACH
 
 config SND_SOC_INTEL_SKL_NAU88L25_MAX98357A_MACH
 	tristate "ASoC Audio driver for SKL with NAU88L25 and MAX98357A in I2S Mode"
-	depends on X86_INTEL_LPSS && I2C
+	depends on MFD_INTEL_LPSS && I2C && ACPI
 	select SND_SOC_NAU8825
 	select SND_SOC_MAX98357A
 	select SND_SOC_DMIC
@@ -216,7 +216,7 @@ config SND_SOC_INTEL_SKL_NAU88L25_MAX98357A_MACH
 
 config SND_SOC_INTEL_BXT_DA7219_MAX98357A_MACH
 	tristate "ASoC Audio driver for Broxton with DA7219 and MAX98357A in I2S Mode"
-	depends on X86 && ACPI && I2C
+	depends on MFD_INTEL_LPSS && I2C && ACPI
 	select SND_SOC_DA7219
 	select SND_SOC_MAX98357A
 	select SND_SOC_DMIC
@@ -230,7 +230,7 @@ config SND_SOC_INTEL_BXT_DA7219_MAX98357A_MACH
 
 config SND_SOC_INTEL_BXT_RT298_MACH
 	tristate "ASoC Audio driver for Broxton with RT298 I2S mode"
-	depends on X86 && ACPI && I2C
+	depends on MFD_INTEL_LPSS && I2C && ACPI
 	select SND_SOC_RT298
 	select SND_SOC_DMIC
 	select SND_SOC_HDAC_HDMI
@@ -244,7 +244,6 @@ config SND_SOC_INTEL_BXT_RT298_MACH
 config SND_SOC_INTEL_BXT_TDF8532_MACH
         tristate "ASoC Audio driver for BXT with TDF8532 in I2S mode"
         depends on X86 && ACPI && I2C
-        select SND_SOC_INTEL_SST
         select SND_SOC_TDF8532
         select SND_SOC_COMPRESS
         help
@@ -254,8 +253,7 @@ config SND_SOC_INTEL_BXT_TDF8532_MACH
  
 config SND_SOC_INTEL_KBL_RT5663_MAX98927_MACH
 	tristate "ASoC Audio driver for KBL with RT5663 and MAX98927 in I2S Mode"
-	depends on X86_INTEL_LPSS && I2C
-	select SND_SOC_INTEL_SST
+	depends on MFD_INTEL_LPSS && I2C && ACPI
 	select SND_SOC_RT5663
 	select SND_SOC_MAX98927
 	select SND_SOC_DMIC
@@ -268,8 +266,8 @@ config SND_SOC_INTEL_KBL_RT5663_MAX98927_MACH
 
 config SND_SOC_INTEL_KBL_RT5663_RT5514_MAX98927_MACH
         tristate "ASoC Audio driver for KBL with RT5663, RT5514 and MAX98927 in I2S Mode"
-        depends on X86_INTEL_LPSS && I2C && SPI
-        select SND_SOC_INTEL_SST
+        depends on MFD_INTEL_LPSS && I2C && ACPI
+        depends on SPI
         select SND_SOC_RT5663
         select SND_SOC_RT5514
         select SND_SOC_RT5514_SPI
@@ -292,7 +290,6 @@ config SND_SOC_INTEL_CNL_CS42L42_MACH
         tristate "SOC Machine Audio driver for CNL Platform"
 	depends on X86 && ACPI && I2C
         select SND_SOC_CS42L42
-	select SND_SOC_INTEL_SST
 	select SND_SOC_DMIC
         help
 	This adds support for ASoC machine driver for cnl and CS42L42. This
@@ -303,7 +300,6 @@ config SND_SOC_INTEL_CNL_RT700_MACH
         tristate "SOC Machine Audio driver for CNL Platform"
 	depends on X86 && ACPI && I2C
 	select SND_SOC_RT700
-	select SND_SOC_INTEL_SST
 	select SND_SOC_DMIC
         help
         This adds support for ASoC machine driver . This will create an alsa
@@ -315,7 +311,6 @@ config SND_SOC_INTEL_CNL_SVFPGA_MACH
         tristate "SND_SOC_INTEL_CNL_SVFPGA_MACH for CNL Platform"
 	depends on X86 && ACPI && I2C
 	select SND_SOC_SVFPGA
-	select SND_SOC_INTEL_SST
 	select SND_SOC_DMIC
         help
 
@@ -323,7 +318,6 @@ config SND_SOC_INTEL_CNL_RT274_MACH
         tristate "ASoC Audio driver for CNL with RT274 in I2S Mode"
 	depends on X86 && ACPI && I2C
 	select SND_SOC_RT274
-	select SND_SOC_INTEL_SST
 	select SND_SOC_DMIC
 	select SND_SOC_HDAC_HDMI
         help
-- 
2.7.4

